### PR TITLE
Allow topics by table name

### PR DESCRIPTION
**Pull request by lileeyao**
@osheroff 

Adding tests now. Just to give a headstart so you know what I'm doing.
### Reference

https://github.com/zendesk/maxwell/issues/286

### CODE REVIEW

**Comment by benmanns** [comment]
You may want to include the database name in addition to the table name for the new topic.

**Comment by lileeyao** [comment]
@osheroff I don't see Kafka related tests, but in local test, we now have (seems fine):
```
[2016-04-12 14:14:21,540] INFO [KafkaApi-0] Auto creation of topic maxwell-maxwell_test_database.test_table with 1 partitions and replication factor 1 is successful! (kafka.server.KafkaApis)

```
@benmanns Sorry for late reply. I've added table name too, but for `pk` thing I'll leave it to @osheroff to decide.

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 1]
```diff
@@ -108,6 +110,10 @@ protected OptionParser buildOptionParser() {
 
 		parser.accepts( "help", "display help").forHelp();
 
+		parser.accepts( "__separator_8" );
+
+		parser.accepts( "use_table_topic", "configure one topic per table instead of single maxwell topic" );
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: put this option in with the kafka options, and name it `kafka_topic_per_table`

File: src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java [hunk 2]
```diff
@@ -93,13 +97,20 @@ public MaxwellKafkaProducer(MaxwellContext context, Properties kafkaProperties,
...
 
 		ProducerRecord<String, String> record =
 				new ProducerRecord<>(topic, this.partitioner.kafkaPartition(r, this.numPartitions), key, value);
 
 		kafka.send(record, new KafkaCallback(r, this.context, key, value));
 	}
 
+	private void addTableTopic(boolean useTableTopic, RowMap r) {
+		if (useTableTopic) {
+			this.topic = this.originalTopic + "-" + r.getDatabase() + "." + r.getTable();
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java]: you don't need to set the instance variable here; it's messy.  better to do something like:

```
   String value = r.toJSON();
   String messageTopic; 
   if ( useTableTopic ) { 
      messageTopic = getTableTopic(...)
  } else {
      messageTopic = this.topic
  }
```

**Comment by osheroff** [comment]
this PR needs to take into consideration:

- the partitioning code needs to be aware of each different topic.
- the default partitioner (by database) probably makes less sense in this use case -- we might want to partition by primary key by default instead. 
- we *may* want to allow both forms of topic names, either with database or without.  

And like we discussed yesterday, it might be that we want to somehow rework all this stuff into a more generic transformation layer rather than doing this featuritis stuff.  But I dunno.

**Comment by benmanns** [comment]
A couple other things I thought of related to this - [Kafka topic name validations](https://github.com/apache/kafka/blob/5236bf60debbb0c08010315a92dd3fbfa482e871/core/src/main/scala/kafka/common/Topic.scala#L24-L25) are different from [MySQL identifier validations](http://dev.mysql.com/doc/refman/5.7/en/identifiers.html) so we may want to strip or sanitize the potential values. E.g.
```sql
CREATE TABLE `foo bar @(*$&(*$ ⛄️` (id INT);
```
actually worked on my local install.

**Comment by osheroff** [comment]
finished elsewhere

**Pull request closed without merging**